src/uraft/uraftcontroller.cc

```diff
@@ -213,6 +229,14 @@ void uRaftController::checkNodeStatus(const boost::system::error_code &error) {
 			}
 			node_alive_ = is_alive;
 		}
+
+		if (!isFloatingIpAlive()) {
```

Comment:
As i mention on previous comment, false, here states floating IP is missing, master service is dead or there was a timeout on the query. Consider logging actual status issue to ping-point exact cause of error from logs.

Reply:
I think possible scenarios are alredy covered in the current version of `isFloatingIpAlive()` function:
```cpp
bool uRaftController::isFloatingIpAlive() {
	std::vector<std::string> params = {"saunafs-uraft-helper",
	                                   "is-floating-ip-alive"};
	std::string result;

	if (runCommand(params, result, opt_.check_cmd_status_period)) {
		if (result == "alive" || result == "dead") { return result == "alive"; }
		syslog(LOG_ERR, "Invalid floating IP status.");
	} else {
		syslog(LOG_WARNING, "is-floating-ip-alive timeout.");
	}

	return false;
}
```